### DISCUSSION TITLE
Support for promotional products / discounts in STOREKIT2_MODE

### DISCUSSION THREAD

**Discussion by Stefanie-Twipe** [hyochan/react-native-iap#2881] [Q&A]
Hi,

I was wondering if somebody could point out to me if there is a way in STOREKIT2_MODE to get promotional offers or discounts like in STOREKIT1_MODE. I found [in documentation](https://react-native-iap.hyo.dev/docs/migrate_to_11.0.0#no-longer-available-in-sk2) that they did not found an equivalent in SK2, however I was wondering if somebody here in the community found a way or made a custom path or ... 

Basically in STOREKIT1_MODE we had this field `discounts` in `SubscriptionIOS` interface, that was filled for promoted products or discounts to the product. However this is not available in STOREKIT2_MODE, or at least it is not filled in this mode.

We were wondering if this is something that will be added in the future or was taken out by design? Since you are still able to add a promoted product in the App Store Connect, it feels weird that this would not be available anymore in STOREKIT2_MODE anyway.

If anybody can shed some light about this topic, I would really appreciate this!

Thanks in advance!

**Comment by hyochan** [comment]
react-native-iap v14+ fully supports StoreKit 2 promotional offers and discounts.

## Available Fields

### Product Level

```typescript
interface ProductSubscriptionIOS {
  // Discount offers array
  discountsIOS?: DiscountIOS[];

  // Subscription info with offers
  subscriptionInfoIOS?: {
    introductoryOffer?: SubscriptionOfferIOS;
    promotionalOffers?: SubscriptionOfferIOS[];
    subscriptionGroupId: string;
    subscriptionPeriod: SubscriptionPeriodValueIOS;
  };
}
```

### Offer Types

```typescript
type SubscriptionOfferTypeIOS = 'introductory' | 'promotional';

interface SubscriptionOfferIOS {
  id?: string;
  paymentMode: string;
  period: SubscriptionPeriodValueIOS;
  periodCount: number;
  price: number;
  type: SubscriptionOfferTypeIOS;
}
```

## Applying a Discount Offer

When purchasing, use `withOffer` parameter:

```typescript
import { requestPurchase } from 'react-native-iap';

await requestPurchase({
  request: {
    apple: {
      sku: 'subscription_id',
      withOffer: {
        identifier: 'offer_id',       // From App Store Connect
        keyIdentifier: 'key_id',      // Your key ID
        nonce: 'unique_nonce',        // UUID string
        signature: 'signature',       // Server-generated signature
        timestamp: Date.now(),
      },
    },
  },
  type: 'subs',
});
```

## Checking Introductory Offer Eligibility

```typescript
import { isEligibleForIntroOfferIOS } from 'react-native-iap';

const isEligible = await isEligibleForIntroOfferIOS('subscription_group_id');
```

## Note on Version

You mentioned `STOREKIT2_MODE` which suggests you're using v12/v13. In v14+, StoreKit 2 is the **default and only mode** - no need to set any mode.

Please upgrade to v14+ for full StoreKit 2 support:

- [Migration Guide (v13 to v14)](https://hyochan.github.io/react-native-iap/docs/guides/migration-v13-to-v14)

## References

- [Apple: Promotional Offers](https://developer.apple.com/documentation/storekit/in-app_purchase/original_api_for_in-app_purchase/subscriptions_and_offers/implementing_promotional_offers_in_your_app)
- [Apple: StoreKit 2 Offers](https://developer.apple.com/documentation/storekit/product/subscriptionoffer)
